### PR TITLE
align the data structure and add redis as dependency

### PR DESCRIPTION
**Pull request by dongshu2013**

### CODE REVIEW

File: mizu_validator/main.py [hunk 1]
```diff
@@ -18,18 +18,18 @@
...
-def do_classify(job: worker.ClassificationJob):
-    job = queue.enqueue(worker.classification_worker, job)
+def do_classify(job: WorkerJob):
+    job = queue.enqueue(job_worker, job)
     return {"job_id": job.id, "status": "Job enqueued"}
 
 
 def start_dev():
-    uvicorn.run("mizu_validator.main:app", host="0.0.0.0", port=8000, reload=True)
+    uvicorn.run("mizu_validator.main:app", host="0.0.0.0", port=8001, reload=True)
```
Comment (dongshu2013) [inline mizu_validator/main.py:27]: use 8001 so it doesn't conflict with the mizu node service

File: mizu_validator/worker.py [hunk 2]
```diff
@@ -8,27 +9,30 @@
...
-
-def classification_worker(job: ClassificationJob):
-    text = get_decoded_value(job.r2_key)
-    tags = classify(text, V1_EMBEDDING)
-    reply = ClassificationResult(job_id=job.job_id, tags=tags)
-    requests.post(
-        job.config.callback_url,
-        json=jsonable_encoder(reply),
-    )
+    output: str | list[str]
```
Comment (germanop) [inline mizu_validator/worker.py:26]: Why do we accept also a plain string only?

File: mizu_validator/main.py (hunk 1 above)
Comment (germanop) [reply mizu_validator/main.py:27]: Probably this should have been configurable from env

Comment (dongshu2013) [reply mizu_validator/main.py:27]: yeah but we can have default port setting as well. The goal here is that if we start two services at local it won't conflict with each other

File: mizu_validator/worker.py (hunk 2 above)
Comment (dongshu2013) [reply mizu_validator/worker.py:26]: should we assume it always be serialized json with proper schema?

File: mizu_validator/main.py (hunk 1 above)
Comment (germanop) [reply mizu_validator/main.py:27]: Agreed, but this is neither..

File: mizu_validator/worker.py (hunk 2 above)
Comment (germanop) [reply mizu_validator/worker.py:26]: IMO yes

**Pull request merged**